edb/schema/indexes.py

```diff
@@ -589,12 +589,13 @@ def compile_expr_field(
                     singletons=frozenset([subject]),
                     apply_query_rewrites=False,
                     track_schema_ref_exprs=track_schema_ref_exprs,
+                    detached=True,
                 ),
             )
 
             # Check that the inferred cardinality is no more than 1
             if expr.irast.cardinality.is_multi():
-                raise errors.ResultCardinalityMismatchError(
+                raise errors.SchemaDefinitionError(
```

Comment:
I've changed this because when running `edgedb migration create`, the CLI would print a stacktrace instead of a nice error. Does it have any other effect?

Reply:
No, I don't think so